notes/fully-determined-protocols.md

```diff
@@ -0,0 +1,41 @@
+# Fully-Determined Protocols
+
+A _fully-determined_ protocol contains all the information to specify the states that should be exchanged as part of the protocol.
```

Comment:
I think "deterministic" describes the notion you're trying to capture, so we should just use that. Otherwise, "fully-determined" sounds like some special type of algorithm.

It's not quite true that such a protocol contains all the information needed to determine which states to exchange, since some of the information will be stored in the "channel store". For example, `OpenChannel` might abort if the nonce has already been used.

Reply:
> I think "deterministic" describes the notion you're trying to capture, so we should just use that. Otherwise, "fully-determined" sounds like some special type of algorithm.

Since none of our protocols satisfy [this](https://en.wikipedia.org/wiki/Deterministic_algorithm#Formal_definition), I think deterministic might be misleading.